### ISSUE TITLE
Question about fine-tuning cINN

### ISSUE THREAD

**Issue by kevin-xuan** [CompVis/image2video-synthesis-using-cINNs#5]
Hello, I would like to fine-tune the cINN model, so I add the scripts on the bair dataset:
```
cINN_model_path = opt.First_stage_model['model_path'] + 'stage2' + '/'
network.flow.load_state_dict(torch.load(cINN_model_path + 'cINN.pth')['state_dict'])
```
It just re-trains the flow module of cINN, after training one epoch, I should observe that PFVD is close to 99.3 in Table 4 in your paper. 
However, I get 288 PFVD. Moreover, the code cannot provide the test_loader, so I test the result on traj_0_to_255.
Is there something I'm doing wrong? The training loggging is as follows:
<img width="835" alt="捕获" src="https://user-images.githubusercontent.com/87954969/206150726-32845495-e491-406f-9220-809f1d1e389e.PNG">

**Comment by kevin-xuan** [comment]
Moreover, I evaluate the FVD on the bair dataset by the eval _synthesis_quality.py and get **103.7** FVD score, which is inconsistent with the reported **99.3** FVD score in Table 4.
<img width="535" alt="捕获" src="https://user-images.githubusercontent.com/87954969/206656758-235dd038-1c9a-410f-b98a-85f7125c1ca7.PNG">
Could you provide a idea how to solve the two problems?

**Comment by mdorkenw** [comment]
Hi! Thanks for your message.
Regarding your first question: during training we used a PyTorch FVD (PFVD) score for logging only which uses a **PyTorch** I3D backbone. The original FVD which is reported in the paper is evaluated using the **Tensorflow** I3D based on the [original script](https://github.com/google-research/google-research/tree/master/frechet_video_distance). The feature space between those two backbones is different, at least to a scaling factor that contributes to the difference in PFVD and FVD.

Regarding the second question: Could you please check that you used the same environment including Cuda 11.1 as described in the repo? Before uploading the code, I checked that the scripts match exactly the numbers reported. Since the model's output is stochastic, meaning sampling with different seeds (as \nu is drawn from N(0,1)) will lead to different outcomes (and by that FVD scores), it is essential to use the settings as described in the repo to obtain the exact score reported. We used a large size of samples for estimating the FVD score to counteract this whenever possible. However, on BAIR the test set size is small which is why the FVD score estimate can be noisy, meaning different seeds/samples will result in slight changes in FVD which explains the discrepancy you experience.

**Comment by kevin-xuan** [comment]
Thanks for your reply. My ubuntu cuda is 11.3, which may be why I get 103.7 FVD.
So, you means that it is normal when PFVD is 288?

**Comment by mdorkenw** [comment]
I don't recall the specific value I got anymore but in the end, it's important that this metric decreases during training as it is a proxy for video quality.

**Comment by kevin-xuan** [comment]
Yeah, the overall fine-tuning process is follows:
<img width="840" alt="捕获" src="https://user-images.githubusercontent.com/87954969/206866468-cc2958a2-d555-4de0-ad8b-b5389b1a6c2d.PNG">
I try to evaluate the FVD score on the test set using the checkpoint_best_val.pth and get 102 FVD score:
<img width="518" alt="捕获1" src="https://user-images.githubusercontent.com/87954969/206867363-99405594-0d56-41e0-bf98-1ed424fea6e7.PNG">

**Comment by kevin-xuan** [comment]
If I would like to fine-tune cINN on another dataset, should I just fine-tune the cINN model or fine-tune the AE and cINN models?
could you give some advice? I really appreciate any help you can provide.

**Comment by mdorkenw** [comment]
I have not tried fine-tuning a model (cINN or AE) on dataset Y when it was originally trained on dataset X. Since the datasets in the paper are quite distinct, I think that it won't work too well. Still, if one wants to try that I believe fine-tuning both would be better.

**Comment by kevin-xuan** [comment]
Thanks for your advice!
